examples/custom-matrix-format/build.sh

```diff
@@ -8,27 +8,20 @@ fi
 BUILD_DIR=$1
 THIS_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" &>/dev/null && pwd )
 
-# copy libraries
-LIBRARY_DIRS="core core/device_hooks reference omp cuda hip"
-LIBRARY_NAMES="ginkgo ginkgo_reference ginkgo_omp ginkgo_cuda ginkgo_hip"
-SUFFIXES=".so .dylib .dll d.so d.dylib d.dll"
-for prefix in ${LIBRARY_DIRS}; do
-    for name in ${LIBRARY_NAMES}; do
-        for suffix in ${SUFFIXES}; do
-            cp ${BUILD_DIR}/${prefix}/lib${name}${suffix} \
-                ${THIS_DIR}/lib${name}${suffix} 2>/dev/null
-        done
-    done
-done
+source ${THIS_DIR}/../build-setup.sh
+
+CXX="nvcc"
 
 # figure out correct compiler flags
 if ls ${THIS_DIR} | grep -F "libginkgo." >/dev/null; then
-    LINK_FLAGS="-lginkgo -lginkgo_omp -lginkgo_cuda -lginkgo_reference -lginkgo_hip"
+    LINK_FLAGS="-lginkgo -lginkgo_omp -lginkgo_cuda -lginkgo_reference -lginkgo_hip -Xlinker -rpath -Xlinker ${THIS_DIR}"
 else
-    LINK_FLAGS="-lginkgod -lginkgo_ompd -lginkgo_cudad -lginkgo_referenced -lginkgo_hipd"
+    LINK_FLAGS="-lginkgod -lginkgo_ompd -lginkgo_cudad -lginkgo_referenced -lginkgo_hipd -Xlinker -rpath -Xlinker ${THIS_DIR}"
+fi
+if [ -z "${CXX}" ]; then
```

Comment:
This is necessary ?

Reply:
Same as all the other examples, but for them the changes are somewhat hidden in the `build-setup.sh`. Without the rpath (and if Ginkgo is not installed!), the examples don't run without setting the `LD_LIBRARY_PATH` to contain the current directory

Reply:
I mean the `CXX` variable. You set it to `nvcc` a few lines above.

Reply:
Ah right, I just copy-pasted that part without thinking. Thanks!